Dataset download zip email

klpoland opened the pull request:
[SFDS-186](https://crc-dev.atlassian.net/browse/SFDS-186)

- Added celery worker and mailhog to docker compose
- refactored celery configs
- Use smtp email backend on local
- Added get_dataset_files and download_file helper functions to avoid duplicate code, easier maintenance
- Added celery task for zipping and emailing files form dataset as well as some test tasks
- Added tests for celery, file download, command to run celery tests with easy to read logs
- Added download button, JS to run DatasetDownloadView
- `make test`, `make test-integration`, and `make pre-commit` pass locally

lucaspar commented on gateway/sds_gateway/api_methods/helpers/download_file.py:
```diff
@@ -0,0 +1,71 @@
+    file_content = None
+
+    try:
+        # Get the file content from MinIO
+        minio_response = client.get_object(
+            bucket_name=settings.AWS_STORAGE_BUCKET_NAME,
+            object_name=target_file.file.name,
+        )
+        file_content = minio_response.read()
+        logger.info("Successfully retrieved file: %s", target_file.file.name)
```
and even then, it does make sense to check for the size; better to prevent a file from being included than to run out of storage.

klpoland replied:
Also, in this case, should I clean up those downloaded files afterward so they don't clutter up the disk?

lucaspar commented on gateway/sds_gateway/api_methods/tasks.py:
```diff
@@ -0,0 +1,189 @@
+            to=[user_email],
+        )
+
+        email.attach(zip_filename, zip_buffer.getvalue(), "application/zip")
```
there's also a case to be made about not recreating the zip if it already exists for a given dataset, but this could be a future PR. For example, multiple "download" clicks while the link is still valid.

klpoland replied:
<img width="1511" alt="Screenshot 2025-06-20 at 8 06 01 AM" src="https://github.com/user-attachments/assets/ec017874-a2a4-4047-a4cc-fc31dba57e91" />

klpoland commented:
- Added an extra confirmation screen
- Send user to a download page link (check login and ownership of dataset to make sure correct person is downloading)
- Added celery flower to compose

lucaspar commented:
> maybe run this at some regular interval, probably not super often. I'm open to suggestions on how to handle the zip files when they expire.

maybe running a [celery's periodic task](https://docs.celeryq.dev/en/stable/userguide/periodic-tasks.html) once a day?
